### ISSUE TITLE
Embed the git revision into citus

### ISSUE THREAD

**Issue by lithp** [citusdata/citus#1936]
It's occasionally useful to know exactly which version of Citus we're playing with. It would be nice to be able to do something like:

```sql
> SELECT citus.source_revision();
f0851257fabc7b2114521b83a97321839d7a742f
```

**Comment by jasonmp85** [comment]
@lithp, I added a `--with-extra-version` flag into Citus recently. If you set up `configure` to add the revision there, it'll show up in `citus_version()`'s output.

This is in fact already performed by packaging for nightlies and custom builds (i.e. anything but an official release). See citusdata/packaging@092a3b512d7e876fc893a0713f970746ac5a9fe1 for details.

**Comment by marcocitus** [comment]
Fixed by #3246
